test/Swashbuckle.AspNetCore.Newtonsoft.Test/SchemaGenerator/NewtonsoftSchemaGeneratorTests.cs

```diff
@@ -88,6 +90,18 @@ public void GenerateSchema_GeneratesFileSchema_IfFormFileOrFileResultType(Type t
             Assert.Equal(expectedEnumCount, schema.Enum.Count);
         }
 
+        [Fact]
+        public void GenerateSchema_DedupsEnumValues_IfEnumTypeWithDuplicateValues()
+        {
+            var schemaRepository = new SchemaRepository();
+
+            var referenceSchema = Subject().GenerateSchema(typeof(HttpStatusCode), schemaRepository);
+
+            Assert.NotNull(referenceSchema.Reference);
+            var schema = schemaRepository.Schemas[referenceSchema.Reference.Id];
+            Assert.Equal(schema.Enum.Cast<OpenApiInteger>().Select(v => v.Value).Distinct().Count(), schema.Enum.Count);
```

Comment:
Because we're making a _dictionary_ from enum member to converted value, we are forced to do the deduplication earlier, in the two separate contract resolvers. Therefore this test is now needed here to check that it's being done right, and won't cause a duplicate key exception.

Reply:
👍